apps/base-docs/base-camp/docs/hardhat-profiling-size/hardhat-profiling-size.md

```diff
@@ -0,0 +1,477 @@
+
+Calling external contracts is explored more deeply in the [Contract to contract interaction](../hardhat-contract-to-contract/hardhat-contract-to-contract.md) article.
+
+But in this example, there is a process that must call both `Calculator` and `Printer`:
+
+```typescript
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import "hardhat/console.sol";
+
+import "./Calculator.sol";
+import "./Printer.sol";
+
+contract Computer {
+    Calculator calculator;
+    Printer printer;
+
+    constructor(address _calculator, address _printer) {
+        calculator = Calculator(_calculator);
+        printer = Printer(_printer);
+    }
+
+    function executeProcess() external view {
+        // call Calculator contract, i.e calculator.add(a, b);
+        // call Printer contract, i.e printer.print("value to print");
+    }
+}
+```
+
+If you run the contract sizer plugin, you get:
+
+```
+ ·------------------------|--------------------------------|--------------------------------·
+ |  Solc version: 0.8.18  ·  Optimizer enabled: true       ·  Runs: 10000                   │
+ ·························|································|·································
+ |  Contract Name         ·  Deployed size (KiB) (change)  ·  Initcode size (KiB) (change)  │
+ ·························|································|·································
+ |  console               ·                 0.084 (0.000)  ·                 0.138 (0.000)  │
+ ·························|································|·································
+ |  Computer              ·                 0.099 (0.000)  ·                 0.283 (0.000)  │
+ ·························|································|·································
+ |  Calculator            ·                 0.751 (0.000)  ·                 0.782 (0.000)  │
+ ·························|································|·································
+ |  Printer               ·                 0.761 (0.000)  ·                 0.792 (0.000)  │
+ ·························|································|·································
+ |  ScientificCalculator  ·                 1.175 (0.000)  ·                 1.206 (0.000)  │
+ ·------------------------|--------------------------------|--------------------------------·
+```
+
+Notice how your `Computer` contract is very small but still has the capability to access all the functionality of `Printer` and `Calculator`.
+
+Although this will reduce the size of each contract, the costs of this are discussed more deeply in the [Gas Optimization lesson](TODO).
```

Comment:
@latechwriter not your fault, but it's failing because of this.  Can you please link this to the hardhat-profiling-gas article and move all the links to the bottom using relative style links?

Reply:
Sure thing!